openeo_processes_dask/process_implementations/cubes/_filter.py

```diff
@@ -15,9 +16,40 @@ def filter_bbox(data: RasterCube, extent: BoundingBox) -> RasterCube:
 
 
 def filter_temporal(
-    data: RasterCube, extent: TemporalInterval, dimension: str
+    data: RasterCube, extent: TemporalInterval, dimension: str = None
 ) -> RasterCube:
-    raise NotImplementedError()
+    temporal_dims = data.openeo.temporal_dims
+
+    if dimension is not None:
+        if dimension not in data.dims:
+            raise DimensionNotAvailable(
+                f"A dimension with the specified name: {dimension} does not exist."
+            )
+        applicable_temporal_dimension = dimension
```

Comment:
Should we raise an exception here if `applicable_temporal_dimension` is not in `temporal_dims`? Or let the user override this, but at least log a warning?

Reply:
Looks good, thanks!